src/rules.cc

```diff
@@ -367,5 +367,26 @@ extern "C" int msc_rules_cleanup(Rules *rules) {
 }
 
 
+extern "C" int msc_rules_reopen_audit_log(Rules *rules, const char **error) {
+    bool succeeded = true;
+    std::string errorStr;
+
+    if (rules->m_auditLog != NULL) {
+        succeeded = rules->m_auditLog->reopen(&errorStr);
+    }
+
+    if (!succeeded) {
+        if (!errorStr.empty()) {
+            *error = strdup(errorStr.c_str());
+        } else {
+            // Guarantee an error message is always assigned in the event of a failure
+            *error = strdup("Unknown error reopening audit log");
+        }
+    }
+
+    return succeeded ? 0 : -1;
```

Comment:
I was looking at `msc_rules_merge`, `msc_rules_add_remote`, `msc_rules_add_file`, and `msc_rules_add` as examples, and they appear to consider negative values as errors.

Oddly, `msc_rules_cleanup` is closest to this function in the source file, and I don't remember thinking about its return value.

Personally, I prefer returning a boolean, but since more functions return a negative value for error, it seems like we should keep it as-is. What do you think?